frameworks/projects/Basic/src/main/royale/org/apache/royale/html/beads/ErrorImage.as

```diff
@@ -0,0 +1,151 @@
+     *  cannot be loaded.
+     * 
+     *  It will be supported by controls that load the ImageModel bead and 
+     *  implement the IImage interface
+	 *  
+	 *  @langversion 3.0
+	 *  @playerversion Flash 10.2
+	 *  @playerversion AIR 2.6
+	 *  @productversion Royale 0.9.8
+	 */
+
+    public class ErrorImage implements IBead {
+
+        protected var _strand:IStrand;
+
+        public function ErrorImage() {            
+        }
+
+        private var _src:String;
+		/**
+		 *  The source of the image
+		 */
+        public function get src():String {
+            return _src;
+        }
+
+        public function set src(value:String):void {
+            _src = value;
+        }
+        
+		COMPILE::JS{
+        private var _hostElement:HTMLImageElement;
+		protected function get hostElement():HTMLImageElement
+		{
+			return _hostElement;
+		}
+        }
+        protected function get hostModel():IImageModel
+        {             
+            return _strand.getBeadByType(IImageModel) as IImageModel;
+        }
+
+        /**
+         *  @copy org.apache.royale.core.IBead#strand
+         *
+         *  @langversion 3.0
+         *  @playerversion Flash 10.2
+         *  @playerversion AIR 2.6
+         *  @productversion Royale 0.9.8
+         */
+        public function set strand(value:IStrand):void 
+        {
+            _strand = value;
+
+	        COMPILE::JS {
+
+                if(_strand is IImage)
+                {
+                    _hostElement = (_strand as IImage).imageElement as HTMLImageElement;
+                    if(_hostElement){
+                        
+                        _hostElement.addEventListener('error', errorHandler);
+
+                        if(_emptyIsError)
+                        {
+                            (_strand as EventDispatcher).addEventListener("beadsAdded", beadsAddedHandler);
+                        }
+                    }   
+                }
+            }
+        }
+
+		COMPILE::JS
+        private function beadsAddedHandler(event:Event):void
+        {
+            (_strand as EventDispatcher).removeEventListener("beadsAdded", beadsAddedHandler);
+
+            if(hostModel)
+                hostModel.addEventListener("urlChanged",srcChangedHandler);
+            srcChangedHandler(null);
+        }
+
+        private var _emptyIsError:Boolean = false;
+        public function get emptyIsError():Boolean {
+            return _emptyIsError;
+        }
+        public function set emptyIsError(value:Boolean):void {
+            _emptyIsError = value;
+        }
+
+		COMPILE::JS
+        private function errorHandler(event:Event):void {
+        
+            if (hostElement.src != _src)
+            {
+                hostElement.src = _src;
+            }
+        }
+		
+        private function srcChangedHandler(event:Event):void
+        {
+            if(hostModel && !hostModel.url){
+                
+                COMPILE::JS {
```

Comment:
COMPILE::JS blocks are generally used for platform specific operations. In this case it doesn't look like there's anything js specific in the block.